source/assets/components/RecipeCard.js

```diff
@@ -0,0 +1,273 @@
+      return value !== undefined;
+    }
+  });
+  return value;
+}
+
+/**
+ * Extract the URL from the given recipe schema JSON object
+ * @param {Object} data Raw recipe JSON to find the URL of
+ * @returns {String} If found, it returns the URL as a string, otherwise null
+ */
+function getUrl(data) {
+  if (data.url) return data.url;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Article') return data['@graph'][i]['@id'];
+    }
+  };
+  return null;
+}
+
+/**
+ * Similar to getUrl(), this function extracts the organizations name from the
+ * schema JSON object. It's not in a standard location so this function helps.
+ * @param {Object} data Raw recipe JSON to find the org string of
+ * @returns {String} If found, it retuns the name of the org as a string, otherwise null
+ */
+function getOrganization(data) {
+  if (data.publisher?.name) return data.publisher?.name;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Organization') {
+        return data['@graph'][i].name;
+      }
+    }
+  };
+  return null;
+}
+
+/**
+ * Converts ISO 8061 time strings to regular english time strings.
+ * Not perfect but it works for this lab
+ * @param {String} time time string to format
+ * @return {String} formatted time string
+ */
+function convertTime(time) {
+  let timeStr = '';
+
+  // Remove the 'PT'
+  time = time.slice(2);
+
+  let timeArr = time.split('');
+  if (time.includes('H')) {
+    for (let i = 0; i < timeArr.length; i++) {
+      if (timeArr[i] == 'H') return `${timeStr} hr`;
+      timeStr += timeArr[i];
+    }
+  } else {
+    for (let i = 0; i < timeArr.length; i++) {
+      if (timeArr[i] == 'M') return `${timeStr} min`;
+      timeStr += timeArr[i];
+    }
+  }
+
+  return '';
+}
+
+/**
+ * Takes in a list of ingredients raw from imported data and returns a neatly
+ * formatted comma separated list.
+ * @param {Array} ingredientArr The raw unprocessed array of ingredients from the
+ *                              imported data
+ * @return {String} the string comma separate list of ingredients from the array
+ */
+function createIngredientList(ingredientArr) {
+  let finalIngredientList = '';
+
+  /**
+   * Removes the quantity and measurement from an ingredient string.
+   * This isn't perfect, it makes the assumption that there will always be a quantity
+   * (sometimes there isn't, so this would fail on something like '2 apples' or 'Some olive oil').
+   * For the purposes of this lab you don't have to worry about those cases.
+   * @param {String} ingredient the raw ingredient string you'd like to process
+   * @return {String} the ingredient without the measurement & quantity 
+   * (e.g. '1 cup flour' returns 'flour')
+   */
+  function _removeQtyAndMeasurement(ingredient) {
+    return ingredient.split(' ').splice(2).join(' ');
+  }
+
+  ingredientArr.forEach(ingredient => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/cse110-fa21-group16/cse110-fa21-group16/pullRequest?prid=8394473)
```suggestion
  ingredientArr.forEach((ingredient) => {
```